### DISCUSSION TITLE
Camera.forward_vector()

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#9988] [Scripting]
### Add a **Camera.foward_vector()**

The whole point of engines is abstraction and encapsulation. If I need a 3 year mathmatics degree to calculate the forward vector of a camera in Godot then something has gone wrong.

Give me a Camera.forward_vector() please!

(This should be the vector of the direction the camera is pointing towards. I know about basis - see 3 year mathmatics degree comment above. And before you ask, they don't teach shearing vectors in 10th grade, so can I just get a camera.forward_vector() and end this once and for all thanks!). If this is too hard to create then how am I as a civilian, expected to calculate this?!?

**Comment by AThousandShips** [comment]
> If I need a 3 year mathmatics degree to calculate the forward vector of a camera in Godot then something has gone wrong.

You dont, you just need basic geometry of the engine, like so:
```gdscript
-my_camera.global_basis.z
```
Or
```gdscript
my_camera.global_basis * Vector3.FORWARD
```
Or if you want local:
```gdscript
-my_camera.basis.z
```
Or
```gdscript
my_camera.basis * Vector3.FORWARD
```

It's also in the documentation [here](https://docs.godotengine.org/en/stable/tutorials/3d/using_transforms.html#obtaining-information)

**Reply by Calinou** [reply]
> It's also in the documentation [here](https://docs.godotengine.org/en/stable/tutorials/3d/using_transforms.html#obtaining-information)

I wonder if this warrants a note in the Camera3D class description too (or perhaps Node3D), considering how commonly this is needed.

**Reply by AThousandShips** [reply]
There is information on it in `Transform3D`, I'm not sure we should duplicate things everywhere and `Transform3D` is clearly visible as a link in `Node3D`
